tests/pytorch/test_optim.py

```diff
@@ -45,6 +45,86 @@ def test_sparse_adam():
     # Pytorch sparseAdam maintains a global step
     # DGL sparseAdam use a per embedding step
 
+@unittest.skipIf(os.name == 'nt', reason='Do not support windows yet')
+@pytest.mark.parametrize('use_uva', [False, True, None])
+def test_sparse_adam_uva(use_uva):
+    if F.ctx().type == 'cpu' and use_uva == True:
+        # we want to only test values of False and None when not using GPU
+        pytest.skip("UVA cannot be used without GPUs.")
+
+    num_embs = 10
+    emb_dim = 4
```

Comment:
We also need to test emb_dim = 1

Reply:
I parameterized the emb_dim in a0de13d to [1,4,101,1024].